include/kamping/distributed_graph_communicator.hpp

```diff
@@ -140,73 +141,80 @@ class CommunicationGraphView {
+    /// @tparam OutNeighborsRange Range type of out-going neighbors.
+    /// @param in_neighbors Range of in-going neighbors.
+    /// @param out_neighbors Range of out-going neighbors.
     ///
-    template <typename InEdgeRange, typename OutEdgeRange>
-    CommunicationGraph(InEdgeRange const& in_edges, OutEdgeRange const& out_edges) {
-        constexpr bool are_in_edges_weighted  = internal::are_edges_weighted<InEdgeRange>();
-        constexpr bool are_out_edges_weighted = internal::are_edges_weighted<OutEdgeRange>();
+    template <typename InNeighborsRange, typename OutNeighborsRange>
+    DistributedCommunicationGraph(InNeighborsRange const& in_neighbors, OutNeighborsRange const& out_neighbors) {
+        constexpr bool are_in_neighbors_weighted  = internal::are_neighborhoods_weighted<InNeighborsRange>();
+        constexpr bool are_out_neighbors_weighted = internal::are_neighborhoods_weighted<OutNeighborsRange>();
         static_assert(
-            are_in_edges_weighted == are_out_edges_weighted,
-            "Weight status of in and out edges is different!"
+            are_in_neighbors_weighted == are_out_neighbors_weighted,
+            "If weighted neighborhoods are passed, they must be provided for both in and out neighbors!"
         );
-        if constexpr (are_in_edges_weighted) {
-            auto get_rank = [](auto const& edge) {
+        auto get_rank = [&](auto const& edge) {
+            if constexpr (are_in_neighbors_weighted) {
                 auto const& [rank, _] = edge;
                 return static_cast<int>(rank);
-            };
+            } else {
+                return static_cast<int>(edge);
+            }
+        };
+
+        transform_elems_to_integers(in_neighbors, _in_ranks, get_rank);
+        transform_elems_to_integers(out_neighbors, _out_ranks, get_rank);
+
+        if constexpr (are_in_neighbors_weighted) {
             auto get_weight = [](auto const& edge) {
                 auto const& [_, weight] = edge;
                 return static_cast<int>(weight);
             };
-            transform_elems_to_integers(in_edges, _in_ranks, get_rank);
-            transform_elems_to_integers(out_edges, _out_ranks, get_rank);
             _in_weights  = DefaultContainer<int>{};
             _out_weights = DefaultContainer<int>{};
-            transform_elems_to_integers(in_edges, _in_weights.value(), get_weight);
-            transform_elems_to_integers(out_edges, _out_weights.value(), get_weight);
-        } else {
-            auto get_rank = [](auto const& edge) {
-                return static_cast<int>(edge);
-            };
-            transform_elems_to_integers(in_edges, _in_ranks, get_rank);
-            transform_elems_to_integers(out_edges, _out_ranks, get_rank);
+            transform_elems_to_integers(in_neighbors, _in_weights.value(), get_weight);
+            transform_elems_to_integers(out_neighbors, _out_weights.value(), get_weight);
```

Comment:
I would still prefer just inlining `std::transform`, instead of having to jump to the definition of a single-line function definition.